app/controllers/admin/createEvents.py

```diff
@@ -19,15 +24,19 @@ def createEvent():
         eventData = request.form.copy() # request.form returns a immutable dict so we need to copy to make changes
         newEventData= setValueForUncheckedBox(eventData)
 
-        #if an event is not recurring then it wil have same end and start date
+        #if an event is not recurring then it will have same end and start date
         if newEventData['eventEndDate'] == '':
             newEventData['eventEndDate'] = newEventData['eventStartDate']
 
-
         # convert date into datetime object (Y-m-d) for the backend
-        newEventData['eventStartDate'] = parser.parse(newEventData['eventStartDate'], dayfirst=True)
-        newEventData['eventEndDate'] = parser.parse(newEventData['eventEndDate'], dayfirst=True)
-
+        newEventData['eventStartDate'] = parser.parse(newEventData['eventStartDate'])
+        newEventData['eventEndDate'] = parser.parse(newEventData['eventEndDate'])
+
+        for entry in range(1,len(newEventData)):

```

Comment:
I don't quite understand this for loop here or in `createNewEvent`. This is looping for each key/value pair in the dictionary, right? Which is not related to the number of new events they want to create.

What you should be doing (it seems to me) is looping through the array from `calculateRecurringEventFrequency`, because that has exactly the number of entries that you are needing to create.

Actually, you could get away with not including `weekX` keys in the event data at all, because you can just regenerate them with `calculateRecurringEventFrequency`